### ISSUE TITLE
Is bonfire able to be used as a local emulator?

### ISSUE THREAD

**Issue by someone1** [kurin/blazer#72]
Using a local emulator to test B2 against would be great - I noticed you made `pyre` and `bonfire` - they seem to be somewhat functional!

What's the story behind the effort here and its plans for the future? The only other project I noticed was [this](https://github.com/mattwelke/B2Emulator) and it doesn't really seem all that useful yet.

**Comment by kurin** [comment]
Yeah, that was the intention.  I ended up not having a ton of time and never finished it.

The (probably overengineered) idea was that different aspects of B2 functionality could be swapped out by different interfaces, so that you could have a FS proxy, or an S3 proxy, or multiple FS proxies, or whatever.  Or a ramdisk proxy if you just wanted to test clients, which was my original impetus.  I might pick it up again but until now I thought I was the only consumer.

**Comment by someone1** [comment]
I'd be happy to lend a hand for some basic usage - maybe we can start by implementing the bare minimum and not worry about all the frills until more needs arise? Something like what [fake-gcs-server](https://github.com/fsouza/fake-gcs-server) does by really only [supporting](https://github.com/fsouza/fake-gcs-server/blob/master/internal/backend/storage.go) bucket creation/listing and, object upload/download/listing/patching.

**Comment by kurin** [comment]
Agreed, I'd love to get this done.  I'm in the middle of about 4 large non-coding projects right now unfortunately but some of them should be finished fairly soon and maybe I can get back to this.  In the meantime if you can get a handle on what's already there (I completely forget) I can take patches.
